genestack_client/settings/genestack_user.py

```diff
@@ -13,24 +13,39 @@
 from getpass import getpass
 from urllib.parse import urlsplit
 
+import requests
+
 from genestack_client import Connection, GenestackAuthenticationException, GenestackException
 from genestack_client.utils import isatty, interactive_select
 
 DEFAULT_HOST = 'platform.genestack.org'
 
 
 def _get_server_url(host):
-    has_scheme = bool(urlsplit(host).scheme)
+    split_host = urlsplit(host)
+    has_scheme = bool(split_host.scheme)
 
     # compatibility with dev settings
     # TODO Add code that will do migration in configs and remove this check.
     if host.startswith('localhost'):
         return 'http://%s/frontend/endpoint' % host
 
-    if has_scheme:
-        return '%s/endpoint' % host
-    else:
-        return 'https://%s/endpoint' % host
+    # default to HTTPS if not explicitly defined
+    url_stub = host if has_scheme else 'https://%s' % host
+    # trust provided path if it ends with '/frontend'
+    if split_host.path.rstrip('/').split('/')[-1] == 'frontend':
```

Comment:
> I suggest that we check url_stub here instead of split_host.path, because I think it is not right that we check one variable but then use a different one.

True, that was a miss on my part.

> Hypothetically url_stub can end differently from split_host.path, e.g. when host contains query parameters.

If `host` contains _query_ (or _fragment_, for that matter), we're screwed anyway, its support has to be implemented explicitly =)